base/docs/basedocs.jl

```diff
@@ -1637,9 +1637,36 @@ julia> ex = ErrorException("I've done a bad thing");
 julia> ex.msg
 "I've done a bad thing"
 ```
+
+!!! compat "Julia 1.12"
+    ErrorException should no longer be used to raise exceptions for invalid field access. Use FieldError instead.
+
 """
 ErrorException
 
+"""
+    FieldError(type::DataType, field::Symbol)
+
+An operation tried to access invalid `field` of `type`.
```

Comment:
`FieldError` is currently only thrown from `getfield`, right? Probably it'd make sense to mention (and xref) `getfield`, like this:

```md
[`getfield`](@ref)
```

Reply:
I agree, but I also had doubts. For consistency with other Exceptions I didn't add it. I will let @LilithHafner to comment on it. I am ok either way.

Reply:
I agree that we should reference `getfield`.